### PR TITLE
Feature/gh 12160

### LINKED ISSUE #12160: Implement order grid (desktop version) UI/UX for for the PDP page

### PR DESCRIPTION
**Pull request by dydome**
close #12160

### CODE REVIEW

File: feature-libs/product/multi-d/components/order-grid/order-grid-counter/order-grid-counter.component.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+export class OrderGridCounterComponent implements OnInit, OnDestroy {
+  @Input() variant: GridVariantOption;
+
+  @Output()
+  quantityChange$ = new EventEmitter<OrderGridEntry>();
+
+  maxQuantity: number = 0;
+  hasStock = false;
+
+  addEntryForm = new FormGroup({
```
Comment (psmul) [inline feature-libs/product/multi-d/components/order-grid/order-grid-counter/order-grid-counter.component.ts]: For me `addEntryForm` suggest more method name rather variable. Can we change it to simply `form` or something more descriptive in terms of form functionality?

File: feature-libs/product/multi-d/components/order-grid/order-grid-counter/order-grid-counter.component.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    this.subscription.add(
+      this.addEntryForm
+        .get('quantity')
+        ?.valueChanges.subscribe((quantity: number) => {
+          this.quantityChange$.emit({
+            quantity,
+            product: this.variant,
+          } as OrderGridEntry);
```
Comment (psmul) [inline feature-libs/product/multi-d/components/order-grid/order-grid-counter/order-grid-counter.component.ts]: No need to type cast as we already provided type above along with declaration

File: feature-libs/product/multi-d/components/order-grid/order-grid-counter/order-grid-counter.component.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        })
+    );
+  }
+
+  private setStockInfo(product: Product): void {
+    this.hasStock =
+      product.stock && product.stock.stockLevelStatus !== 'outOfStock'
+        ? true
+        : false;
```
Comment (psmul) [inline feature-libs/product/multi-d/components/order-grid/order-grid-counter/order-grid-counter.component.ts]:
```suggestion
      product.stock && product.stock.stockLevelStatus !== 'outOfStock'
```

I think we can shorten it this way

File: feature-libs/product/multi-d/components/order-grid/order-grid.component.ts [hunk 4]
```diff
@@ -1,10 +1,63 @@
...
+import { Observable } from 'rxjs';
+import {
+  GridVariantOption,
+  VariantsMultiDimensionalService,
+} from '../../core/services/variants-multi-dimensional.service';
+
+export interface OrderGridEntry {
+  quantity?: number;
+  product: Product;
+}
```
Comment (psmul) [inline feature-libs/product/multi-d/components/order-grid/order-grid.component.ts]: can we move it to models folder?

File: feature-libs/product/multi-d/components/order-grid/order-grid.component.ts [hunk 5]
```diff
@@ -1,10 +1,63 @@
 import { Component, ChangeDetectionStrategy } from '@angular/core';
+import { Product } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import {
+  GridVariantOption,
+  VariantsMultiDimensionalService,
+} from '../../core/services/variants-multi-dimensional.service';
```
Comment (psmul) [inline feature-libs/product/multi-d/components/order-grid/order-grid.component.ts:12]: We can't have relative imports. This ones should be imported from `@spartacus/product/multi-d/core`

**Review by psmul — CHANGES_REQUESTED** [review]

File: feature-libs/product/multi-d/components/order-grid/order-grid.component.ts [hunk 6]
```diff
@@ -1,10 +1,63 @@
...
+      (element: OrderGridEntry) => element.product.code === entry.product.code
+    );
+
+    if (entryIndex === -1) {
+      const element = {
+        quantity: entry.quantity,
+        product: {
+          code: entry.product.code,
+        },
+      } as OrderGridEntry;
```
Comment (psmul) [inline feature-libs/product/multi-d/components/order-grid/order-grid.component.ts]:
```suggestion
      const element: OrderGridEntry = {
        quantity: entry.quantity,
        product: {
          code: entry.product.code,
        },
      };
```

File: feature-libs/product/multi-d/components/order-grid/order-grid.component.ts (hunk 5 above)
Comment (dydome) [reply feature-libs/product/multi-d/components/order-grid/order-grid.component.ts:12]: I will fix all imports in the end of epic work

**Pull request closed without merging**